### PR TITLE
Bugs and Improvements for CPP SDK and Example

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

This fixes a bug in the CPP SDK, that was highlighted in the CPP example, but didn't end show up in the conformance tests, as well as several quality of life improvements.

* Bring all the gRPC references to our current gRPC version in the project.
* Add `-j$(nproc)` for all `build` targets to utlise more cores when compiling. Shaved some time off compilation.
* Make the setup and compilation of the example align with the unit tests (where it makes sense) - so same base image, etc.
* Explicitly compile the Abseil dependency (this was the critical issue!)

**Which issue(s) this PR fixes**:

Work on #3281

**Special notes for your reviewer**:

Not quite sure if there is a good way to force this kind of issue to show up in the conformance tests, given that we rely on gRPC to be installed on the base image all SDK images are derived from -- but at least this gets us over the current hurdle.

### CODE REVIEW

**Review by gongmax — APPROVED** [review]

**Comment by markmandel** [comment]
@Kalaiselvi84 once this is merged, this should unblock you on completing #3281 👍🏻

**Comment by markmandel** [comment]
```
CMake Error at CMakeLists.txt:32 (_message):
  Some (but not all) targets in this export set were already defined.

  Targets Defined:
  gRPC::re2;gRPC::address_sorting;gRPC::gpr;gRPC::grpc;gRPC::grpc_unsecure;gRPC::grpc++;gRPC::grpc++_alts;gRPC::grpc++_error_details;gRPC::grpc++_reflection;gRPC::grpc++_unsecure;gRPC::grpc_plugin_support;gRPC::grpcpp_channelz;gRPC::upb


  Targets not yet defined: gRPC::grpc_authorization_provider

Call Stack (most recent call first):
  .build/.install/gRPC/lib/cmake/grpc/gRPCTargets.cmake:37 (message)
  .build/.install/gRPC/lib/cmake/grpc/gRPCConfig.cmake:23 (include)
  CMakeLists.txt:77 (find_package)


-- Configuring incomplete, errors occurred!
See also "/go/src/agones.dev/agones/sdks/cpp/.build/CMakeFiles/CMakeOutput.log".
See also "/go/src/agones.dev/agones/sdks/cpp/.build/CMakeFiles/CMakeError.log".
make: *** [Makefile:29: build] Error 1
make[1]: *** [includes/sdk.mk:88: run-sdk-command] Error 2
make: *** [includes/sdk.mk:72: run-sdk-command-cpp] Error 2
```

This is passing locally, so I'm very confused.

**Pull request merged**
